lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -192,15 +193,15 @@ def add_buyer(post, payment_method, options)
           buyer[:dniType] = buyer_hash[:dni_type]
           buyer[:cnpj] = buyer_hash[:cnpj] if options[:payment_country] == 'BR'
           buyer[:emailAddress] = buyer_hash[:email]
-          buyer[:contactPhone] = options[:shipping_address][:phone] if options[:shipping_address]
+          buyer[:contactPhone] = (options[:billing_address][:phone] if options[:billing_address]) || (options[:shipping_address][:phone] if options[:shipping_address])
```

Comment:
I don't _think_ this is doing what you intend; the result of this line as-is will be `buyer[:contactPhone]` set to `options[:billing_address][:phone]`, `options[:shipping_address][:phone]`, or `nil`, whereas I suspect you meant to leave it unset entirely if neither is available.

I'm not aware of a clever way to short-circuit this in our version of Ruby. The closest I could get would be something like

    val = (options[:billing_address] || options[:shipping_address] || {})[:phone]
    order[:contactPhone] = val if val

but I'm not sure that level of clever is actually wise.

Reply:
What she has is actually what is desired, at least according to the ticket. Though looking at the ticket I'm getting a sinking sense that my summary of what should be done is not actually what the customer wants. I'll come back to this.

Reply:
I didn't see the attached ticket, so that may well be the case. I'll yoink my reject.

Reply:
In fact, after reviewing the ticket again this is in fact what is basically desired, however it is likely best to send an empty string in the case of neither field being available.